src/app/api/admin/services/externalLogin/route.ts

```diff
@@ -11,49 +17,69 @@ export async function POST(req: NextRequest) {
 
   try {
     const payload = await req.json();
-    const { email, password } = payload;
-    const user = await db.user.findFirst({
-      where: {
-        email,
-      },
+
+    const result = loginSchema.safeParse(payload);
+    if (!result.success) {
+      return NextResponse.json(
+        { message: 'Invalid input', errors: result.error.errors },
+        { status: 400 },
+      );
+    }
+
+    const { email, password } = result.data;
+
+    const user = await db.user.findUnique({
+      where: { email },
       select: {
         id: true,
         email: true,
         name: true,
         password: true,
       },
     });
-    if (!user) {
-      return NextResponse.json({ message: 'User not found' }, { status: 404 });
+
+    // Perform a dummy bcrypt compare if user doesn't exist to prevent timing attacks
+    if (!user || !user.password) {
```

Comment:
Can do make another pr with just this change and nothing else.

Reply:
Done